locations/spiders/petrol_bg.py

```diff
@@ -12,4 +13,18 @@ def parse_item(self, item, location):
         if m := re.match(r"^(\d+) (.+)$", item["name"]):
             item["ref"] = m.group(1)
             item["name"] = m.group(2)
+
+        categories = (location["categories"] or "").split(",")
+        apply_yes_no(Fuel.DIESEL, item, ("19" in categories or "20" in categories))
+        apply_yes_no(Fuel.OCTANE_100, item, "21" in categories)
+        apply_yes_no(Fuel.OCTANE_95, item, "22" in categories)
+        apply_yes_no(Fuel.LPG, item, "23" in categories)
+        apply_yes_no(Fuel.CNG, item, "24" in categories)
+        apply_yes_no(Extras.CAR_WASH, item, "25" in categories)
+        apply_yes_no(Extras.ATM, item, "26" in categories)
+        apply_yes_no("restaurant", item, "27" in categories)
```

Comment:
Ideally immediately after `categories=` the following may be possible:

```
if "27" in categories:
  restaurant_item = copy.deepcopy(item)
  apply_category(Categories.RESTAURANT, item)
  yield item
```

This will create a separate item with the same address, phone number, etc. Perhaps some fields need to be pop'd off (such as phone number) if they differ between fuel station and restaurant. Perhaps a different brand needs to be applied, etc.

Reply:
They have two kinds of restaurants. One is the typical petrol station restaurant (sandwiches, hot dogs, cafe). The other is an actual restaurant called "Street Chefs" which is available on very few Petrol station (probably less than 10). I will check back on it in the coming days when I have a bit more time.

Reply:
The stations that have `restaurant` actually have cafe which at best sells some baked snacks and has outdoor seating area. The ones that have actual restaurant ("Street Chefs") are only two. `cafe=yes` would probably be a better tag.

Reply:
Updated the PR to use `cafe=yes` instead.